### PR TITLE
Set up oauth2

### LINKED ISSUE #9: Set up oauth2
## Why

For API authentication

## Acceptance Criteria
- [ ]  Integrate oauth2 into the project

### PR DESCRIPTION
**Pull request by junan**
https://github.com/junan/google-scraper/projects/1#card-50542051

## What happened 👀

Set up oauth2

## Insight 📝

Used [oauth2](https://github.com/go-oauth2/oauth2)

## Proof Of Work 📹

Screen Shot 2021-03-30 at 12 51 17 PM
Screen Shot 2021-03-30 at 11 45 53 AM

### CODE REVIEW

File: initializers/oauth_server.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package initializers
```
Comment (junan) [inline initializers/oauth_server.go:1]: This file code mostly copied from the oauth2 doc

File: services/oauth/client.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+func GenerateOauthClient(userID int64) (client *models.Client, err error) {
+	clientId := uuid.New().String()
+	clientSecret := uuid.New().String()
+
+	client = &models.Client{
+		ID:     clientId,
+		Secret: clientSecret,
+		UserID: IntToString(userID),
```
Comment (olivierobert) [inline services/oauth/client.go:17]: Why is the ID stored as a string? 🤔 Is this column be default provided by the oauth2 library?

Comment (junan) [reply services/oauth/client.go:17]: Yes, this column provided by [Oauth2](https://github.com/go-oauth2/oauth2/blob/master/models/client.go#L8) and its type also string

Comment (Lahphim) [inline services/oauth/client.go:17]: @junan What's your purpose of using this `UserID`? 🤔
Is this use for filtering for just only client belonging to that user?

Comment (Lahphim) [reply services/oauth/client.go:17]: Junan will keep this for his future task 🙌

**Review by Lahphim — APPROVED** [review]

**Review by gutakk — APPROVED** [review]

**Pull request merged**